src/connection/services/Vcard.ts

```diff
@@ -17,11 +18,129 @@ export default class Vcard extends AbstractService {
       return this.sendIQ(iq).then(this.parseVcard);
    }
 
+   public setAvatar(jid: IJID, avatar: string, mimetype: string)
+   {
+      let getVcardIQ = $iq({
+         type: 'get',
+         to: jid.bare
+      }).c('vCard', {
+         xmlns: NS.get('VCARD')
+      });
+
+      //first get the actual vcard to merge image into
+      return this.sendIQ(getVcardIQ).then((stanza) => {
```

Comment:
Would it not be easier to use `loadVcard`?

Reply:
loadVcard calls parseVcard which will call parseVcardChildren and does not return a stanza in which we could merge the photo